impl/src/metadata.rs

```diff
@@ -516,9 +515,7 @@ pub mod tests {
 
   impl DummyCargoMetadataFetcher {
     fn render_metadata(&self, mock_workspace_path: &Path) -> Option<Metadata> {
-      if self.metadata_template.is_none() {
-        return None;
-      }
+      self.metadata_template.as_ref()?;
```

Comment:
Can you store this to a value and use it later on line 532?  That'll remove both the unwrap in those lines, and make the intent of this more clear.